florist/app/jobs/details/page.tsx

```diff
@@ -164,33 +173,51 @@ export function JobDetailsStatus({ status }: { status: string }): ReactElement {
     );
 }
 
-export function JobProgress({
-    serverMetrics,
-    serverConfig,
+export function JobProgressBar({
+    metrics,
+    totalEpochs,
     status,
 }: {
-    serverMetrics: string;
-    serverConfig: string;
+    metrics: string;
+    totalEpochs: number;
     status: status;
 }): ReactElement {
     const [collapsed, setCollapsed] = useState(true);
 
-    if (!serverMetrics || !serverConfig) {
+    if (!metrics || !totalEpochs) {
         return null;
     }
 
-    const serverMetricsJson = JSON.parse(serverMetrics);
-    const serverConfigJson = JSON.parse(serverConfig);
+    const metricsJson = JSON.parse(metrics);
+
+    let endRoundKey;
+    if (metricsJson.type === "server") {
+        endRoundKey = "fit_end";
+    }
+    if (metricsJson.type === "client") {
+        endRoundKey = "shutdown";
+    }
 
     let progressPercent = 0;
-    if ("rounds" in serverMetricsJson && Object.keys(serverMetricsJson.rounds).length > 0) {
-        const totalServerRounds = serverConfigJson.n_server_rounds;
-        const lastRound = Math.max(...Object.keys(serverMetricsJson.rounds));
-        const lastCompletedRound = "fit_end" in serverMetricsJson ? lastRound : lastRound - 1;
-        progressPercent = (lastCompletedRound * 100) / totalServerRounds;
+    if ("rounds" in metricsJson && Object.keys(metricsJson.rounds).length > 0) {
+        const lastRound = Math.max(...Object.keys(metricsJson.rounds));
+        const lastCompletedRound = endRoundKey in metricsJson ? lastRound : lastRound - 1;
+        progressPercent = (lastCompletedRound * 100) / totalEpochs;
     }
     const progressWidth = progressPercent === 0 ? "100%" : `${progressPercent}%`;
 
+    // Clients will not have a status, so we need to set one based on the progress percent
+    if (!status) {
+        if (progressPercent === 0) {
+            status = "NOT_STARTED";
+        } else if (progressPercent === 100) {
+            status = "FINISHED_SUCCESSFULLY";
+        } else {
+            status = "IN_PROGRESS";
+        }
+        // TODO: add error status
```

Comment:
I gather we are trying to set status when it does not exist based on the progressPercent. Just curious, What do you intend on adding with the: // TODO: add error status? Is there ever a time where we would set to FINISHED_WITH_ERROR based on the value of progressPercent? I guess if its not a number or outside range of 0 to 100?

Reply:
We need to add errors to the metrics so we can reliably check them here. We could do stuff like you proposed but it's unreliable, I prefer to try to do it the right way and if not possible we can come up with some heuristics to try to find that out. Thanks for raising that, I will create tasks in clickup to track it.

Reply:
Makes sense, definitely on the same page. I was just confused how you can get the error status from the progressPercent. But looks like this is what the TODO was getting at. Figure out how to catch FINISHED_WITH_ERROR